addons/sale/controllers/catalog.py

```diff
@@ -0,0 +1,112 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+from odoo.tools import groupby
+
+
+class CatalogController(Controller):
+
+    @route('/sales/catalog/sale_order_lines_info', auth='user', type='json')
+    def sale_product_catalog_get_sale_order_lines_info(self, order_id, product_ids):
+        """ Returns products information to be shown in the catalog.
+
+        :param int order_id: The sale order, as a `sale.order` id.
+        :param list product_ids: The products currently displayed in the product catalog, as a list
+                                 of `product.product` ids.
+        :rtype: dict
+        :return: A dict with the following structure:
+            {
+                'productId': int
+                'quantity': float (optional)
+                'price': float
+                'readOnly': bool (optional)
+            }
+        """
+        order = request.env['sale.order'].browse(order_id)
+        sale_order_line_info = []
+        for product, lines in groupby(
+            order.order_line.filtered(lambda line: not line.display_type),
+            lambda line: line.product_id
+        ):
+            if (len(lines) > 1):
```

Comment:
Sooo, if i understand correctly:
If the product_id is not in the requested product_ids but there are multiple lines with the product, you'll load the information, but not if it's only present in one line ???